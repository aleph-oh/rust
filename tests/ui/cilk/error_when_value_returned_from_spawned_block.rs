#![feature(cilk)]
// Tests that compilation fails when a value is returned from a spawned block before
// a sync runs.

fn main() {
    let x = { 
        let y = cilk_spawn { 1 };
        y
//~^ ERROR: used binding `y` is possibly-uninitialized [E0381]
    };
}