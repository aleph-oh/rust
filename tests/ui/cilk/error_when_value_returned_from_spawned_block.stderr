error[E0381]: used binding `y` is possibly-uninitialized
  --> $DIR/error_when_value_returned_from_spawned_block.rs:7:9
   |
LL |         let y = cilk_spawn { 1 };
   |             -              ----- binding initialized here in some conditions
   |             |
   |             binding declared here but left uninitialized
LL |         y
   |         ^ `y` used here but it is possibly-uninitialized

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0381`.
